### DISCUSSION TITLE
Why there is multiple package.json?

### DISCUSSION THREAD

**Discussion by edbrdi** [tamagui/tamagui#822] [Q&A] [answered]
I was wondering why there is multiple package.json (root, apps/expo, apps/next, packages/app, packages/ui...) and not only one at the root of the project?

For example, if I need to add i18n-js, do I add it at the root package.json? What is the rule to know where to add dependencies?

**Comment by Nadjem-47** [comment, marked as answer]
Because the template is a monorepo , means that it is a single repository containing multiple distinct projects, with pre defined relationships .

The main benefit of this is sharing packages and components between multiple projects in one single main repository .

The rules of adding dependencies is simple :

1 - If you have shared dependency add it at the root level .
2 - If it's used only by one project just add it there .

**Reply by edbrdi** [reply]
Thanks for the explanation!
